packages/generator/templates/app/app/auth/mutations/resetPassword.ts

```diff
@@ -0,0 +1,51 @@
+import { SessionContext } from "blitz"
+import db from "db"
+import { ResetPasswordInput, ResetPasswordInputType } from "../validations"
+import { hashToken, hashPassword } from "../auth-utils"
+
+export class ResetPasswordError extends Error {
+  name = "ResetPasswordError"
+  message = "Reset password link is invalid or it has expired."
+}
+
+export default async function resetPassword(
+  input: ResetPasswordInputType,
+  _ctx: { session?: SessionContext } = {}
+) {
+  const { password, token } = ResetPasswordInput.parse(input)
+
+  // 1. Try to find this token in the database
+  // Have to first decodeURIComponent(token) because encodeURIComponent
+  // was used in forgotPassword.ts when sending to the browser
+  const hashedToken = hashToken(decodeURIComponent(token))
+  const possibleToken = await db.token.findOne({
+    where: { hashedToken, type: "RESET_PASSSWORD" },
```

Comment:
This line should be changed to account for the fact that there's a unique index on hashedToken and type:
`    where: { hashedToken_type: { hashedToken, type: 'RESET_PASSSWORD' } },`